x/uibc/uics20/memo_handler.go

```diff
@@ -71,50 +69,52 @@ func (mh MemoHandler) dispatchMemoMsgs(ctx *sdk.Context, receiver sdk.AccAddress
 }
 
 // error messages used in validateMemoMsg
-const (
-	msg0typeErr = "only MsgSupply, MsgSupplyCollateral and MsgLiquidate are supported as messages[0]"
-	msg1typeErr = "only MsgBorrow is supported as messages[1]"
+var (
+	errNoSubCoins = errors.New("message must use only coins sent from the transfer")
+	errMsg0Type   = errors.New("only MsgSupply, MsgSupplyCollateral and MsgLiquidate are supported as messages[0]")
+	// errMsg1Type = errors.New("only MsgBorrow is supported as messages[1]")
 )
 
 // We only support the following message combinations:
 // - [MsgSupply]
 // - [MsgSupplyCollateral]
-// - [MsgSupplyCollateral, MsgBorrow] -- here, borrow must use
 // - [MsgLiquidate]
 // Signer of each message (account under charged with coins), must be the receiver of the ICS20
 // transfer.
-func (mh MemoHandler) validateMemoMsg(receiver sdk.AccAddress, sent sdk.Coin, msgs []sdk.Msg) error {
+func (mh MemoHandler) validateMemoMsg(_receiver sdk.AccAddress, sent sdk.Coin, msgs []sdk.Msg) error {
 	msgLen := len(msgs)
-	if msgLen > 2 {
-		return errors.New("ics20 memo with more than 2 messages are not supported")
-	}
-
-	for _, msg := range msgs {
-		if signers := msg.GetSigners(); len(signers) != 1 || !signers[0].Equals(receiver) {
-			return errors.New(
-				"msg signer doesn't match the receiver, expected signer: " + receiver.String())
-		}
+	// In this release we only support 1msg, and only messages that don't create or change
+	// a borrow position
+	if msgLen > 1 {
+		return errors.New("ics20 memo with more than 1 message is not supported")
 	}
 
 	var (
-		asset      sdk.Coin
-		collateral sdk.Coin
+		asset sdk.Coin
+		// collateral sdk.Coin
 	)
 	switch msg := msgs[0].(type) {
 	case *ltypes.MsgSupplyCollateral:
 		asset = msg.Asset
-		collateral = asset
+		// collateral = asset
 	case *ltypes.MsgSupply:
 		asset = msg.Asset
 	case *ltypes.MsgLiquidate:
 		asset = msg.Repayment
 	default:
-		return errors.New(msg0typeErr)
+		return errMsg0Type
 	}
 
-	// TODO more asserts, will be handled in other PR
-	if err := assertSubCoins(sent, asset); err != nil {
-		return err
+	return assertSubCoins(sent, asset)
+
+	/**
+	   TODO: handlers v2
+
+	for _, msg := range msgs {
+		if signers := msg.GetSigners(); len(signers) != 1 || !signers[0].Equals(receiver) {
+			return errors.New(
+				"msg signer doesn't match the receiver, expected signer: " + receiver.String())
+		}
```

Comment:
I think we have to check the signer and receiver are  same if it is one message also

Reply:
We don't need, because we don't allow borrows . In the current situation it's like sending money to someone else account which you can perfectly do with IBC today

Reply:
Okay got it 👍